superset-frontend/src/components/AlteredSliceTag/AlteredSliceTag.test.jsx

```diff
@@ -17,325 +17,164 @@
+  expect(result).toEqual('Min: 1, Max: 2');
+});
 
-    it('returns Metrics if the field type is metrics', () => {
-      const value = [
-        {
-          label: 'SUM(Sales)',
-        },
-      ];
-      const expected = 'SUM(Sales)';
-      expect(
-        wrapper.instance().formatValue(value, 'metrics', controlsMap),
-      ).toBe(expected);
-    });
+test('formatValueHandler formats CollectionControl values correctly', () => {
+  const value = [{ a: 1 }, { b: 2 }];
+  const result = formatValueHandler(value, 'key3', controlsMap);
+  expect(result).toEqual(
+    `${JSON.stringify(value[0])}, ${JSON.stringify(value[1])}`,
+  );
+});
 
-    it('stringifies objects', () => {
-      const value = { 1: 2, alpha: 'bravo' };
-      const expected = '{"1":2,"alpha":"bravo"}';
-      expect(
-        wrapper.instance().formatValue(value, undefined, controlsMap),
-      ).toBe(expected);
-    });
+test('formatValueHandler formats MetricsControl values correctly', () => {
+  const value = [{ label: 'Metric1' }, { label: 'Metric2' }];
+  const result = formatValueHandler(value, 'key4', controlsMap);
+  expect(result).toEqual('Metric1, Metric2');
+});
 
-    it('does nothing to strings and numbers', () => {
-      expect(wrapper.instance().formatValue(5, undefined, controlsMap)).toBe(5);
-      expect(
-        wrapper.instance().formatValue('hello', undefined, controlsMap),
-      ).toBe('hello');
-    });
+test('formatValueHandler formats boolean values correctly', () => {
+  const value = true;
+  const result = formatValueHandler(value, 'key5', controlsMap);
+  expect(result).toEqual('true');
+});
 
-    it('returns "[]" for empty filters', () => {
-      expect(
-        wrapper.instance().formatValue([], 'adhoc_filters', controlsMap),
-      ).toBe('[]');
-    });
+test('formatValueHandler formats array values correctly', () => {
+  const value = [{ label: 'Label1' }, { label: 'Label2' }];
+  const result = formatValueHandler(value, 'key5', controlsMap);
+  expect(result).toEqual('Label1, Label2');
+});
 
-    it('correctly formats filters with array values', () => {
-      const filters = [
-        {
-          clause: 'WHERE',
-          comparator: ['1', 'g', '7', 'ho'],
-          expressionType: 'SIMPLE',
-          operator: 'IN',
-          subject: 'a',
-        },
-        {
-          clause: 'WHERE',
-          comparator: ['hu', 'ho', 'ha'],
-          expressionType: 'SIMPLE',
-          operator: 'NOT IN',
-          subject: 'b',
-        },
-      ];
-      const expected = 'a IN [1, g, 7, ho], b NOT IN [hu, ho, ha]';
-      expect(
-        wrapper.instance().formatValue(filters, 'adhoc_filters', controlsMap),
-      ).toBe(expected);
-    });
+test('formatValueHandler formats string values correctly', () => {
+  const value = 'test';
+  const result = formatValueHandler(value, 'key5', controlsMap);
+  expect(result).toEqual('test');
+});
 
-    it('correctly formats filters with string values', () => {
```

Comment:
Could we add tests for specific controls like this one? I don't see them in the refactored tests and I think they're valuable

Reply:
This was actually being handled but only for string filters by the AdhocFilter test. It was pretty abstract so I made it more explicit in the revision